mantidproject/mantid
# Plot advanced against most properties

This allows for more easily plotting against a log that was filtered.

**Report to:** Huq

**To test:**

Do a workflow that filters events then right-click on the resulting workspace group and `Plot` -> `Advanced`. You'll see that there are a lot more options, including whatever you filtered on. A script that generates a "useful" workspace is
```python
from mantid.simpleapi import GenerateEventsFilter, Load, SNSPowderReduction

# Load data's log only
Load(Filename='PG3_41603.nxs.h5',
     OutputWorkspace='PG3_41603_meta',
     MetaDataOnly=True)

# Construct the event filters
GenerateEventsFilter(InputWorkspace='PG3_41603_meta',
                     OutputWorkspace='PG3_41603_splitters',
                     InformationWorkspace='PG3_41603_splitinfo',
                     LogName='SampleTemp',
                     MinimumLogValue='10',
                     MaximumLogValue='300',
                     FilterLogValueByChangingDirection='Both',
                     LogValueInterval='2',
                     LogBoundary='Left')

SNSPowderReduction(SaveAs='gsas',
                   FinalDataUnits='dSpacing',
                   Filename='PG3_41603',
                   Sum=False,
                   Binning=-0.0008,
                   OutputDirectory='/tmp',
                   BinInDspace=True,
                   CalibrationFile='/SNS/PG3/shared/CALIBRATION/2018_2_11A_CAL/PG3_MICAS_d42093_2018_10_03.h5',
                   CharacterizationRunsFile='/SNS/PG3/shared/CALIBRATION/2018_2_11A_CAL/PG3_char_2018_07_05.txt,/SNS/PG3/shared/CALIBRATION/2018_2_11A_CAL/PG3_char_2018_06_11-HighRes-PAC.txt',
                   StripVanadiumPeaks=True,
                   RemovePromptPulseWidth=50.0,
                   FilterBadPulses=10.0,
                   PushDataPositive='ResetToZero',
                   BackgroundSmoothParams='5,2',
                   PreserveEvents=True,
                   ScaleData=100.0,
                   SplittersWorkspace='PG3_41603_splitters',
                   SplitInformationWorkspace='PG3_41603_splitinfo')
```

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
